engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActor.scala

```diff
@@ -268,6 +294,27 @@ final case class WorkflowExecutionActor(workflowId: WorkflowId,
       stay() using startRunnableScopes(newData)
   }
 
+  private def pushSuccessfulJobMetadata(jobKey: JobKey, returnCode: Option[Int], outputs: JobOutputs) = {
+    import MetadataServiceActorImplicits.EnhancedServiceRegistryActorForMetadata
+    pushCompletedJobMetadata(jobKey, ExecutionStatus.Done, returnCode)
+    outputs foreach { case (lqn, value) => serviceRegistryActor.pushWdlValueMetadata(metadataKey(jobKey, s"${CallMetadataKeys.Outputs}:$lqn"), value.wdlValue) }
+  }
+
+  private def pushFailedJobMetadata(jobKey: JobKey, returnCode: Option[Int], failure: Throwable, retryableFailure: Boolean) = {
+    import MetadataServiceActorImplicits._
+    pushCompletedJobMetadata(jobKey, ExecutionStatus.Failed, returnCode)
+    serviceRegistryActor.pushThrowableMetadata(metadataKey(jobKey, s"${CallMetadataKeys.Failures}[${failure.hashCode}]"), failure)
```

Comment:
Again, that's in another PR which hasn't affected this branch yet... :)

I would say "race to push" but I guess I could just change it myself...